### PR TITLE
docs: proposal restructure

### PR DESCRIPTION
**Pull request by mffap**
This PR contains a major restructure of our docs pages. Goals are to have a cleaner structure of examples, quickstarts, and SDKs as well as bring together the different tabs for easier navigation.

This will potentially break a couple of external backlinks to the docs. Refactoring of underlying files was not done in this PR to reduce the impact. We can follow up later and setup redirects.

What has changed
- Addresses overviews (Examples, SDKs, Quickstarts) from https://github.com/zitadel/zitadel/issues/5124
- Remove manual/"Help" and integrate in the rest of the docs
- Make docs easier to navigate by combining the side bars
- Create an integration section, replacing https://github.com/zitadel/zitadel/pull/4554
- Clean-up solution scenarios
- Add section Manage > Users
- Split Architecture from Concepts; Eventstore remains part of Architecture

Todos:

- [x] Check for internal broken links (docusaurus warnings)
- [ ] Align folders/files with the new structure
- [x] Update links and structure on main page /docs
- [ ] Remove orphaned pages

```[tasklist]
### Definition of Ready
- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] Where possible E2E tests are implemented
- [ ] Critical parts are tested automatically
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.
```

### CODE REVIEW

**Comment by mffap** [comment]
@dakshitha @fforootd @hifabienne the initial proposal for the restructured docs. You can find an overview of changes in the Description. I would appreciate your initial thoughts and review.

There are still some clean-up tasks to do, that's why I leave the PR in draft for now.

**Comment by fforootd** [comment]
Left some thoughts here https://vercel.live/open-feedback/docs-git-mpaexamples20230227-zitadel.vercel.app?nextHash=show-comment-archive%3Dtrue

Overall I really like the structure really

**Comment by mffap** [comment]
> Left some thoughts here https://vercel.live/open-feedback/docs-git-mpaexamples20230227-zitadel.vercel.app?nextHash=show-comment-archive%3Dtrue
> 
> Overall I really like the structure really

I prefer the oldschool comments in code, I just realized. Already lost the overview of comments 😅

**Comment by fforootd** [comment]
> > Left some thoughts here https://vercel.live/open-feedback/docs-git-mpaexamples20230227-zitadel.vercel.app?nextHash=show-comment-archive%3Dtrue
> > Overall I really like the structure really
> 
> I prefer the oldschool comments in code, I just realized. Already lost the overview of comments 😅

I will take a not of this 😁

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -65,14 +65,14 @@ module.exports = {
...
           docId: "guides/overview",
           position: "left",
-        },
+        }, 
         {
           type: "doc",
-          label: "Examples",
-          docId: "examples/introduction",
+          label: "Get Started",
+          docId: "guides/start/quickstart",
```
Comment (fforootd) [inline docs/docusaurus.config.js:75]: This is what I meant confuses me (that two nav links point to two different anchors in the same root page.

But i know now why I was triggered, by the fact that the "Quick Start Guide" is a single item under the "Get Started" Section. 😁 

I think a quick fix would be to move the content to the menu item "Get started"

**Comment by fforootd** [comment]
I converted my comments into GH comments 😁

File: docs/sidebars.js [hunk 2]
```diff
@@ -1,54 +1,65 @@
...
       ],
-      collapsed: false,
-    },
-    {
-      type: "category",
-      label: "Secure your API",
-      items: ["examples/secure-api/go", "examples/secure-api/python-flask", "examples/secure-api/dot-net"],
-      collapsed: false,
     },
+    "examples/sdks",
```
Comment (fforootd) [inline docs/sidebars.js:12]: I think we should move this below quickstarts or into integrate

Comment (mffap) [reply docs/sidebars.js:12]: Yes makes sense and pulls together the get started section(s)

File: docs/sidebars.js [hunk 3]
```diff
@@ -1,54 +1,65 @@
...
+        {
+          type: "category",
+          label: "Frontend",
+          items: [
+            "examples/login/angular",
+            "examples/login/react",
+            "examples/login/flutter",
+            "examples/login/nextjs",
+          ],
+          collapsed: false,
```
[the remark this answers is not in the document]
Comment (mffap) [reply docs/sidebars.js:26]: Interestingly this one wasn't bugging me :) But I get the point. I would propose to have the main sections collapsed to get a first overview and then expand the sub-sections. I would make the exception for the Get Started, to have it jump out.

File: docs/docusaurus.config.js (hunk 1 above)
Comment (mffap) [reply docs/docusaurus.config.js:75]: Yes combining the sections makes sense and looks good. Having the get started on top as CTA was an idea we wanted to test.

File: docs/sidebars.js [hunk 4]
```diff
@@ -84,86 +95,172 @@ module.exports = {
...
+            "guides/integrate/services/gitlab-saml",
+            "guides/integrate/services/auth0-oidc",
+            "guides/integrate/services/auth0-saml",
+            "guides/integrate/services/pingidentity-saml",
           ],
         },
         {
           type: "category",
-          label: "SAML 2.0 Clients",
+          label: "Infrastructure",
```
[the remark this answers is not in the document]
Comment (mffap) [reply docs/sidebars.js:186]: Sure, let's try it.

File: docs/sidebars.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mffap) [reply docs/sidebars.js:26]: Ok. I think this is better now.

**Comment by mffap** [comment]
I think we should merge it first without changing the underlying file structure. This causes less issues with redirects etc. we can take the file restructure as follow-up task.

**Review by fforootd — APPROVED** [review]

**Comment by fforootd** [comment]
This looks good to me!

**Comment by mffap** [comment]
@eliobischof do you have any veto or feedback for now?
I would prefer if we merge soon and do tweaks later, except there are some blockers.

File: docs/package.json [hunk 5]
```diff
@@ -4,7 +4,8 @@
   "private": true,
   "scripts": {
     "docusaurus": "docusaurus",
-    "start": "yarn generate && docusaurus start",
+    "start": "docusaurus start",
+    "start:api": "yarn generate && docusaurus start",
```
[the remark this answers is not in the document]
Comment (fforootd) [reply docs/package.json:8]: sure thing

File: docs/docs/concepts/introduction.mdx [hunk 6]
```diff
@@ -2,43 +2,9 @@
 title: Introduction
```
Comment (eliobischof) [inline docs/docs/concepts/introduction.mdx:2]: I see we have content on category pages. Should we consistently move introducing content there? So in this case /docs/concepts/introduction to /docs/concepts?

File: docs/docs/guides/overview.mdx [hunk 7]
```diff
@@ -4,16 +4,22 @@ title: Overview
 
 Most applications need to know the identity of a user for access control, to securely store their data in the cloud, and provide the same personalized experience across all of the user's devices.
 
-ZITADEL provides backend services, easy-to-use SDKs, and ready-made UI libraries to authenticate users in your application. It supports authentication using passwords and applies additional security with the help of a second factor, for example, OTP, to ensure safe and secure access.
-It additionally leverages industry standards like OAuth 2.0 and OpenID Connect such that it can be easily integrated into your custom backend.
+With ZITADEL you can rely on a hardened and extensible turnkey solution to solve all of your authentication and authorization needs.
+We provide you with a wide range of out of the box features to accelerate your project. Multi-tenancy with branding customization, secure login, self-service, OpenID Connect, OAuth2.x, SAML2, Passwordless with FIDO2 (including Passkeys), OTP, U2F, and an unlimited audit trail is there for you, ready to use.
+
```
Comment (eliobischof) [inline docs/docs/guides/overview.mdx:9]:
```suggestion
Execute custom code on selected events within ZITADEL to ensure perfect compatibility with your unique and complex software landscape and data models.

```

File: docs/docs/guides/overview.mdx [hunk 8]
```diff
@@ -29,9 +35,14 @@ Choose [ZITADEL Self-Hosted](/self-hosting/deploy/overview) if you want:
...
 - Flexibility when you deploy updates
 
-:::info
-Support is available either through the opensource community or a support contract.
-:::
+## Get Help
+
+Join our [Discord Chat](https://zitadel.com/chat) or open a [Discussion](https://github.com/zitadel/zitadel/discussions) on Github to get help from the community and the team behind ZITADEL.
+
+Cloud and Enterprise customers can additionally reach us privately via the [Support communication channels](http://localhost:3000/docs/legal/support-services).
```
Comment (eliobischof) [inline docs/docs/guides/overview.mdx:42]: Is it correct like this?

```suggestion
Cloud and Enterprise customers can additionally reach us privately via the [Support communication channels](/legal/support-services).
```

File: docs/docusaurus.config.js [hunk 9]
```diff
@@ -65,16 +65,10 @@ module.exports = {
       items: [
         {
           type: "doc",
-          label: "Guides",
+          label: "Documentation",
```
Comment (eliobischof) [inline docs/docusaurus.config.js:68]: The terminology seems not very precise, but I don't have a better idea.

**Review by eliobischof — CHANGES_REQUESTED** [review]

**Comment by eliobischof** [comment]
@mffap this comment is the only blocker https://github.com/zitadel/zitadel/pull/5318#discussion_r1126720196
The others are more FYI, you can just resolve them.

File: docs/docs/concepts/introduction.mdx (hunk 6 above)
Comment (mffap) [reply docs/docs/concepts/introduction.mdx:2]: I've actually tried it out for the pages which were lacking an overview. I think it's an good idea. We can do this as a follow-up.

File: docs/docusaurus.config.js (hunk 9 above)
Comment (mffap) [reply docs/docusaurus.config.js:68]: Yes, it's all except api reference and self-hosting instructions. Alternatives I've thought of were "Home" or "Product" but was also not happy with it.

**Comment by mffap** [comment]
thanks @eliobischof for the review. let's keep the naming of the top nav item and the category pages as follow-up.

File: docs/docs/guides/overview.mdx (hunk 8 above)
Comment (eliobischof) [reply docs/docs/guides/overview.mdx]: I mean I didn't test the suggested link, but looks good I guess

**Review by eliobischof — APPROVED** [review]

**Pull request merged**